### PR TITLE
Clubs Page PR 1/5: Factored out <ModalAddOrChangeYourClub> and <ModalRemoveYourClub>

### PR DESCRIPTION
**Pull request by mmmavis**
I'm breaking [the original PR](https://github.com/mozilla/teach.webmaker.org/pull/1198) apart since it was too huge to digest.

This is the step#1

- factored out `<ModalAddOrChangeYourClub>` and `<ModalRemoveYourClub>`
- created `/clubs/list` (real content to be filled in later)

### CODE REVIEW

File: components/modal-clubs-remove.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+var React = require('react/addons');
```
Comment (mmmavis) [inline components/modal-clubs-remove.jsx]: everything in this file was extracted from `pages/clubs.jsx`](https://github.com/mozilla/teach.webmaker.org/blob/develop/pages/clubs.jsx)

File: components/modal-clubs.jsx [hunk 2]
```diff
@@ -0,0 +1,295 @@
+var React = require('react/addons');
```
Comment (mmmavis) [inline components/modal-clubs.jsx]: all the code here was extracted from [`pages/clubs.jsx`](https://github.com/mozilla/teach.webmaker.org/blob/develop/pages/clubs.jsx) as well

File: pages/clubs-list.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+var React = require('react');
```
Comment (mmmavis) [inline pages/clubs-list.jsx]: Added basic structure to this new page. Real content were added in succeeding PRs

File: pages/clubs.jsx [hunk 4]
```diff
@@ -2,7 +2,6 @@ var _ = require('underscore');
 var React = require('react/addons');
```
Comment (mmmavis) [inline pages/clubs.jsx]: factored out `<ModalAddOrChangeYourClub>` and `<ModalRemoveYourClub>`

File: test/browser/clubs-page.test.jsx [hunk 5]
```diff
@@ -9,28 +9,9 @@ var StubTeachAPI = require('./stub-teach-api');
 var StubRouter = require('./stub-router');
```
Comment (mmmavis) [inline test/browser/clubs-page.test.jsx]: moved relevant tests to the corresponding files, i.e., `modal-clubs-remove.test.jsx` and 
`modal-clubs.test.js`

**Comment by mmmavis** [comment]
@Pomax this PR is all about factoring existing code. No extra logic nor content were created here (well, except what's in `pages/clubs-list.jsx`)

File: components/modal-clubs.jsx (hunk 2 above)
Comment (Pomax) [inline components/modal-clubs.jsx]: this component is so big... =P

**Comment by Pomax** [comment]
this looks pretty good, tests pass, and the clubs-list is sensible. R+

**Pull request merged**